src/chainstate/stacks/index/proofs.rs

```diff
@@ -281,47 +341,60 @@ impl TrieMerkleProof {
 
         let mut proof = vec![];
         
-        let mut back_block = backptr.back_block();
         let mut block_header = storage.get_cur_block();
 
-        let ancestor_block_hash = storage.block_walk(back_block)?;
-        storage.open_block(&ancestor_block_hash, false)?;
+        let ancestor_block_hash = storage.get_block_from_local_id(backptr.back_block())?.clone();
+        storage.open_block(&ancestor_block_hash)?;
 
         let ancestor_root_hash = read_root_hash(storage)?;
 
         let mut found_backptr = false;
-        
+
+        let ancestor_height = MARF::get_block_height(storage, &ancestor_block_hash, &block_header)?
+            .ok_or_else(|| Error::CorruptionError(format!("Could not find block height of {}", &block_header)))?;
+        let mut current_height = MARF::get_block_height(storage, &block_header, &block_header)?
+            .ok_or_else(|| Error::CorruptionError(format!("Could not find block height of {}", &block_header)))?;
+
+        if current_height == ancestor_height {
+            eprintln!("Already at the ancestor: {} =? {}, heights: {} =? {}", &ancestor_block_hash, &block_header,
```

Comment:
Please replace all instances of `eprintln!()` with the appropriate logging macro (e.g. `debug!`, `info!`, `warn!`, `error!`).

Reply:
Addressed in d31f0207